### PR TITLE
Add agent2systema rpcs for RemoveIntercept

### PR DESCRIPTION
**Pull request by donnyyung**
This adds a gRPC which enables the agent to inform Ambassador Cloud
(System A) that an intercept has been left.

Signed-off-by: Donny Yung <donaldyung@datawire.io>

---

## Helpful information

- CircleCI will run the linters and unit tests on your PR.
  - The results of that CI run will be listed as "check-local."
  - You can run the same tests yourself:

    ```shell
    make lint check-local
    ```

  - The other checks for your PR will succeed immediately without testing anything. They rely on Datawire secrets to push images or talk to a Kubernetes cluster, so they are disabled for forked-repo PRs.

- Please include a changelog entry as a file `newsfragments/issue_number.type`.
  - `type` is one of `incompat`, `feature`, `bugfix`, or `misc`
  - E.g., `1003.bugfix` would contain the text
    > Attaching a debugger to the process running under Telepresence no longer causes the session to end.
  - You can preview the changelog with

    ```shell
    virtualenv/bin/towncrier --draft
    ```

- Please delete this pre-filled text ("Helpful information"). Note that you can edit the description after submitting the PR if you prefer.

Thank you for your contribution!

### CODE REVIEW

File: rpc/systema/agent2systema.proto [hunk 1]
```diff
@@ -30,3 +33,7 @@ message ReviewInterceptResponse {
   // allowed to become active.
   string msg = 2;
 }
+
+message LeftIntercept {
```
Comment (donnyyung) [inline rpc/systema/agent2systema.proto]: I wasn't sure what to name this... Not sure if `LeftInterceptInfo` would be better... naming is hard so open to suggestions.

File: rpc/systema/agent2systema.proto [hunk 2]
```diff
@@ -2,19 +2,22 @@ syntax = "proto3";
 
 package telepresence.systema;
 
+import "google/protobuf/empty.proto";
 import "rpc/manager/manager.proto";
 
 option go_package = "github.com/telepresenceio/telepresence/rpc/v2/systema";
 
-// How Calls from the agent to System A are authenticated is TBD.
+// How Calls from the agent to System A are authenticated is via apikey.
```
Comment (LukeShu) [inline rpc/systema/agent2systema.proto]: This sentence is now awkwardly worded.  The sentence was there for symmetry with the multiple comments in `manager2systema.proto`:
```
// Calls from the X to Y are authenticated via RFC 6750
// Bearer tokens.
```
But as I paste that, I realize that comment is outdated; it isn't using RFC 6750 Bearer tokens any more.  So probably update those comments too?

So maybe "Calls from the X to Y are authenticated via API keys per https://www.notion.so/datawire/Telepresence-System-A-API-Key-Integration-c13c4676e9944abfb02713b8517051e3 (see also: `pkg/client/connector/auth/apikey.go`, `cmd/traffic/cmd/manager/systema.go:systemaCredentials`)."? I know it's a faux pas to link to non-public things, but... no one outside of Ambassador Labs is going to be hacking on auth with Ambassador Cloud, and it's the best reference for us when we do hack on it.

File: rpc/systema/agent2systema.proto (hunk 1 above)
Comment (LukeShu) [reply rpc/systema/agent2systema.proto]: `Left` sounds weird.  Perhaps `RemovedIntercept`?  Or `InterceptRemoval`? I think I like `InterceptRemoval` best.

**Comment by LukeShu** [comment]
The general strategy seems good to me.

**Comment by donnyyung** [comment]
I have went back-and-forth on this all day, but I think it makes sense to do this in the manager for a few reasons:
1. It was a bunch of extra work to see if an intercept had been removed instead of just hooking into the RemoveIntercept Call.
2. When an agent's pod restarts, the intercept is still listed as active (which might be a bug but I think is the behavior we want), so if i put this logic in the agent, then it would say the intercept was removed when it really wasn't.
3. I think my hesitancy to do it in the manager was bc managing the API Keys might be more complicated, but I think it's better to pick the right location for the solution than pick whichever is easier.

**Pull request closed without merging**